src/app/component/SwipeToDelete/SwipeToDelete.component.js

```diff
@@ -0,0 +1,176 @@
+/**
+ * ScandiPWA - Progressive Web App for Magento
+ *
+ * Copyright © Scandiweb, Inc. All rights reserved.
+ * See LICENSE for license details.
+ *
+ * @license OSL-3.0 (Open Software License ("OSL") v. 3.0)
+ * @package scandipwa/base-theme
+ * @link https://github.com/scandipwa/base-theme
+ */
+
+import PropTypes from 'prop-types';
+import { createRef, PureComponent } from 'react';
+
+import Draggable from 'Component/Draggable';
+import { ChildrenType } from 'Type/Common';
+import CSS from 'Util/CSS';
+
+import {
+    ANIMATION_DURATION,
+    DRAG_RIGHT_OPEN_TRESHHOLD,
+    DRAG_RIGHT_OPEN_TRIGGER_THRESHOLD
+} from './SwipeToDelete.config';
+
+import './SwipeToDelete.style';
+
+/** @namespace Component/SwipeToDelete/Component */
+export class SwipeToDelete extends PureComponent {
+    static propTypes = {
+        children: ChildrenType.isRequired,
+        dragRightOpenTriggerThreshold: PropTypes.number,
+        dragRightOpenThreshold: PropTypes.number,
+        animationDuration: PropTypes.number,
+        renderRightSideContent: PropTypes.func,
+        rightSideMix: PropTypes.object,
+        topElemMix: PropTypes.object
+    };
+
+    static defaultProps = {
+        dragRightOpenTriggerThreshold: DRAG_RIGHT_OPEN_TRIGGER_THRESHOLD,
+        dragRightOpenThreshold: DRAG_RIGHT_OPEN_TRESHHOLD,
+        animationDuration: ANIMATION_DURATION,
+        renderRightSideContent: () => {},
+        rightSideMix: {},
+        topElemMix: {}
+    };
+
+    state = {
+        isRightSideOpen: false
+    };
+
+    draggableRef = createRef();
+
+    componentDidMount() {
+        this.setTranlateXStyle(0);
```

Comment:
Please provide verbose comments for this logic. It will take you 5 minutes to comment on your logic in some 10 comments and it will take 2 hours for other developers to understand it if it will be necessary in the future.